rust/agents/relayer/src/msg/gelato_submitter/fwd_req_op.rs

```diff
@@ -1,22 +1,254 @@
-use std::time::Duration;
+use std::{sync::Arc, time::Duration};
 
+use abacus_base::InboxContracts;
+use abacus_core::{ChainCommunicationError, Inbox, InboxValidatorManager, MessageStatus};
+use ethers::{
+    signers::Signer,
+    types::{H160, U256},
+};
 use eyre::Result;
-use gelato::fwd_req_call::ForwardRequestArgs;
+use gelato::{
+    chains::Chain,
+    fwd_req_call::{
+        ForwardRequestArgs, ForwardRequestCall, ForwardRequestCallResult, PaymentType,
+        NATIVE_FEE_TOKEN_ADDRESS,
+    },
+    task_status_call::{TaskStatus, TaskStatusCall, TaskStatusCallArgs},
+};
+use tokio::{
+    sync::mpsc::UnboundedSender,
+    time::{sleep, timeout},
+};
+use tracing::instrument;
+
+use crate::msg::SubmitMessageArgs;
+
+/// The max fee to use for Gelato ForwardRequests.
+/// Gelato isn't charging fees on testnet. For now, use this hardcoded value
+/// of 1e18, or 1.0 ether.
+/// TODO: revisit before running on mainnet and when we consider interchain
+/// gas payments.
+const DEFAULT_MAX_FEE: u64 = 10u64.pow(18);
+
+/// The default gas limit to use for Gelato ForwardRequests, arbitrarily chose
+/// to be 5M.
+/// TODO: once Gelato fully deploys their new version, simply omit the gas
+/// limit so that Gelato does the estimation for us.
+const DEFAULT_GAS_LIMIT: u64 = 5000000;
 
-// TODO(webbhorn): Remove 'allow unused' once we impl run() and ref internal fields.
-#[allow(unused)]
 #[derive(Debug, Clone)]
 pub(crate) struct ForwardRequestOp<S> {
-    pub args: ForwardRequestArgs,
-    pub opts: ForwardRequestOptions,
-    pub signer: S,
-    pub http: reqwest::Client,
+    opts: ForwardRequestOptions,
+    http: reqwest::Client,
+
+    message: SubmitMessageArgs,
+    inbox_contracts: InboxContracts,
+    sponsor_signer: S,
+    sponsor_address: H160,
+    // Currently unused due to a bug in Gelato's testnet relayer that is currently being upgraded.
+    _sponsor_chain: Chain,
+    destination_chain: Chain,
+
+    /// A channel to send the message over upon the message being successfully processed.
+    message_processed_sender: UnboundedSender<SubmitMessageArgs>,
 }
 
-impl<S> ForwardRequestOp<S> {
-    #[allow(unused)]
-    pub async fn run(&self) -> Result<()> {
-        todo!()
+impl<S> ForwardRequestOp<S>
+where
+    S: Signer,
+    S::Error: 'static,
+{
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I would propose we do this:
```
pub struct ForwardRequestOpArgs { ... }; // all public

pub struct ForwardRequestOp(ForwardRequestOpArgs);

impl ForwardRequestOp {
  fn new(args: ForwardRequestOpArgs) -> Self { Self(args) }
  ...
}

// for external; internal uses `self.0`
impl Deref for ForwardRequestOp {
  Target = ForwardRequestOpArgs;
  ...
}
```

Reply:
wdyt of just getting rid of the `fn new` and require `ForwardRequestOp`s to be created with `ForwardRequestOp { ... }` syntax?

Is your suggestion just stylistically different from that or also semantically different?